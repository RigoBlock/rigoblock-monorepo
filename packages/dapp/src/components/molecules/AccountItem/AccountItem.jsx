import './AccountItem.scss'
import { METAMASK } from '../../../constants/user'
import MetaMaskLogo from '../../../images/metamask-fox.svg'
import PropTypes from 'prop-types'
import React from 'react'
import Tooltip, { TOOLTIP_SIZES } from '../../atoms/Tooltip'

export const ACCOUNT_ITEM_TYPES = {
  FULL: 'full',
  SHORT: 'short'
}

const AccountItem = ({ provider, number, appearance }) => {
  const hiddenAccount = `**** ${number.substr(number.length - 4, 4)}`
  let displayedProvider = provider && provider.toLowerCase()
  let logoComponent = (
    <img src={`https://identicon-api.herokuapp.com/${number}/24?format=png`} />
  )

  if (displayedProvider === METAMASK.toLowerCase()) {
    displayedProvider = METAMASK
    logoComponent = <MetaMaskLogo />
  }
  // conditional logic for Parity to be added
  return appearance === ACCOUNT_ITEM_TYPES.SHORT ? (
    <div className="account-item-short">
      <div className="logo">{logoComponent}</div>
      <div className="info">
        <span className="provider">{displayedProvider}</span>
        <span className="account-number">{hiddenAccount}</span>
      </div>
    </div>
  ) : (
    <div className="account-item-full">
      <div className="provider">
        <div className="logo">{logoComponent}</div>
        <span className="name">{displayedProvider}</span>
      </div>
      <span className="account-number">
        <span>{number}</span>
        {/* TODO: add copy to clipboard function */}
        <Tooltip type="filter_none" size={TOOLTIP_SIZES.SMALL} />
      </span>
    </div>
  )
}

AccountItem.propTypes = {
  provider: PropTypes.string.isRequired,
  number: PropTypes.string.isRequired,
  appearance: PropTypes.oneOf([
    ACCOUNT_ITEM_TYPES.FULL,
    ACCOUNT_ITEM_TYPES.SHORT
  ])
}

AccountItem.defaultProps = {
  appearance: ACCOUNT_ITEM_TYPES.SHORT
}

export default AccountItem
