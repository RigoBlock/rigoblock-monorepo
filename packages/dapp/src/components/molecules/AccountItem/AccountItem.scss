@import "../../_settings/variables.scss";

.accountItem {
  display: flex;
  span {
    display: block;
  }
  .account-item-provider {
    font-weight: 700;
  }

  .account-item-logo {
    width: 24px;
    height: 24px;
  }

  > *:not(:last-child) {
    margin-right: $dist-xs;
  }
}
