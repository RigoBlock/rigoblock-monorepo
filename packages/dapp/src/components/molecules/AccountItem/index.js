export { default, ACCOUNT_ITEM_TYPES } from './AccountItem'
