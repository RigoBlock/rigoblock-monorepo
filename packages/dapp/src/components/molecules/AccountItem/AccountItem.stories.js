import '../../_settings/_base.scss'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import AccountItem from './AccountItem'
import React from 'react'

const props = {
  provider: 'metamask',
  number: '0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196'
}

storiesOf('Molecules/AccountItem', module)
  .addDecorator(withKnobs)
  .add('default', () => <AccountItem {...props} />)
