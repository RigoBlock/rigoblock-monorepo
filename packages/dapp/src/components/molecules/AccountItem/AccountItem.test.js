import { shallow } from 'enzyme'
import AccountItem from './AccountItem'
import toJson from 'enzyme-to-json'

const props = {
  provider: 'metamask',
  number: '0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d'
}

describe('AccountItem component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(AccountItem, props)))
    ).toMatchSnapshot()
  })
})
