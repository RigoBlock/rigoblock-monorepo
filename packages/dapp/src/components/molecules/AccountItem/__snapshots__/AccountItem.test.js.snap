// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountItem component renders correctly (full version) 1`] = `
<div
  className="account-item-full"
>
  <div
    className="provider"
  >
    <div
      className="logo"
    >
      <Component />
    </div>
    <span
      className="name"
    >
      MetaMask
    </span>
  </div>
  <span
    className="account-number"
  >
    <span>
      0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d
    </span>
    <TooltipIcon
      onClick={[Function]}
      size="small"
      type="filter_none"
    />
  </span>
</div>
`;

exports[`AccountItem component renders correctly (short version) 1`] = `
<div
  className="account-item-short"
>
  <div
    className="logo"
  >
    <Component />
  </div>
  <div
    className="info"
  >
    <span
      className="provider"
    >
      MetaMask
    </span>
    <span
      className="account-number"
    >
      **** 2d6d
    </span>
  </div>
</div>
`;
